src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.Client/ProjectNodeExtenderCommandHandler.cs

```diff
@@ -0,0 +1,88 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.Input;
+using Microsoft.VisualStudio.ProjectSystem.VS;
+using Microsoft.VisualStudio.RpcContracts.OpenDocument;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Workspace.VSIntegration.UI;
+
+namespace Microsoft.VisualStudio
+{
+    internal class ProjectNodeExtenderCommandHandler : IWorkspaceCommandHandler
+    {
+        private readonly JoinableTaskContext _taskContext;
+        private readonly IServiceProvider _serviceProvider;
+
+        public ProjectNodeExtenderCommandHandler(JoinableTaskContext taskContext, IServiceProvider serviceProvider)
+        {
+            _taskContext = taskContext;
+            _serviceProvider = serviceProvider;
+        }
+
+        public int Priority => 0x0100;
+
+        public bool IgnoreOnMultiselect => true;
+
+        public int Exec(List<WorkspaceVisualNodeBase> selection, Guid pguidCmdGroup, uint nCmdID, uint nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
+        {
+            if (pguidCmdGroup == CommandGroup.ManagedProjectSystemClientProjectEditGuid)
+            {
+                var nCmdIDInt = (int)nCmdID;
+
+                switch (nCmdIDInt)
+                {
+                    case ManagedProjectSystemClientProjectEditCommandId.EditProjectFile:
+                        OpenFile(selection.SingleOrDefault());
+
+                        return HResult.OK;
+                }
+            }
+
+            return HResult.Ole.Cmd.NotSupported;
+        }
+
+        private void OpenFile(WorkspaceVisualNodeBase workspaceVisualNodeBase)
+        {
+            _taskContext.Factory.RunAsync(async () =>
+            {
+                var serviceContainer = _serviceProvider.GetService<SVsBrokeredServiceContainer, IBrokeredServiceContainer>();
+                var serviceBroker = serviceContainer.GetFullAccessServiceBroker();
+
+                var openDocumentService = await serviceBroker.GetProxyAsync<IOpenDocumentService>(VisualStudioServices.VS2019_4.OpenDocumentService);
+
+                try
+                {
+                    await openDocumentService.OpenDocumentWithEditorSelectionAsync(workspaceVisualNodeBase.NodeMoniker, cancellationToken: default);
+                }
+                finally
+                {
+                    (openDocumentService as IDisposable)?.Dispose();
+                }
+            });
+        }
+
+        public bool QueryStatus(List<WorkspaceVisualNodeBase> selection, Guid pguidCmdGroup, uint nCmdID, ref uint cmdf, ref string customTitle)
+        {
+            bool handled = false;
+
+            if (pguidCmdGroup == CommandGroup.ManagedProjectSystemClientProjectEditGuid)
+            {
+                var nCmdIDInt = (int)nCmdID;
+
+                switch (nCmdIDInt)
+                {
+                    case ManagedProjectSystemClientProjectEditCommandId.EditProjectFile:
```

Comment:
Ideally we would only make the item visible if the selected node is a project node. As QueryStatus never seems to be called this seems to be moot at the moment.